src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -160,7 +177,7 @@ export class ElasticsearchConfig {
    */
   public readonly customHeaders: ElasticsearchConfigType['customHeaders'];
 
-  constructor(rawConfig: ElasticsearchConfigType) {
+  constructor(rawConfig: ElasticsearchConfigType, log?: Logger) {
```

Comment:
Since passing the logger to this constructor is temporary (shouldn't be needed when #40255 is resolved), I left the param as optional -- to avoid TypeScript errors in test files and to avoid changing those files.